### PR TITLE
[13.0][FIX] sale_order_type: Inconsistent data in multi-company environments

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

ping @carlosdauden @chienandalu

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

**Comment by rousseldenis** [comment]
@sergio-teruel Could you fix pre-commit ?

**Comment by sergio-teruel** [comment]
@rousseldenis Fixed!, thanks

**Review by Tardo — APPROVED** [review]
Code review

**Comment by sergio-teruel** [comment]
Travis red unrelated to this module.
Is due to collision between sale_order_revision and sale_quotation_number.

File: sale_order_type/models/sale_order_type.py [hunk 1]
```diff
@@ -33,24 +35,27 @@ def default_picking_policy(self):
     journal_id = fields.Many2one(
         comodel_name="account.journal",
         string="Billing Journal",
-        domain=[("type", "=", "sale")],
+        domain="[('type', '=', 'sale'), '|', ('company_id', '=', False), "
+        "('company_id', '=', company_id)]",
```
Comment (chienandalu) [inline sale_order_type/models/sale_order_type.py:39]: When the company isn't set and the journal is, this will fail (can a journal have no company?)

Comment (sergio-teruel) [reply sale_order_type/models/sale_order_type.py:39]: You are right, the journal company_id field is required, but this domain will be not fails.
If you prefer i can change by [('type', '=', 'sale'), ('company_id', '=', company_id)] ...

**Comment by sergio-teruel** [comment]
@pedrobaeza Ready!!

**Comment by pedrobaeza** [comment]
Pushed directly the workaround for the random failing test extracted from here in 02891714e79a841aa484818d9b3b366060101e51, so merging this without it:

/ocabot merge minor

**Comment by pedrobaeza** [comment]
Cherry-picked for v14 in 8a8f9c7cd35208bc2dcbbaf43a0ae45da9b73b02

**Pull request merged**